bitcoin/test/fuzz/miniscript_random.cpp

```diff
@@ -309,10 +310,111 @@ std::optional<NodeInfo> ConsumeNode(FuzzedDataProvider& provider) {
+    // count the number of matching rules up front.
+    int fragcode = provider.ConsumeIntegral<uint8_t>();
+
+    while (true) {
+        /** Counter for the number of applicable rules. */
+        int candidates = 0;
+
+        // Rules for constructing leaf nodes
+        if ("Bzud"_mst << typ && ++candidates && !(fragcode--)) return {{Fragment::JUST_0}};
+        if ("Bzu"_mst << typ && ++candidates && !(fragcode--)) return {{Fragment::JUST_1}};
+        if ("Kondu"_mst << typ && ++candidates && !(fragcode--)) return {{Fragment::PK_K, ConsumePubKey(provider)}};
+        if ("Kndu"_mst << typ && ++candidates && !(fragcode--)) return {{Fragment::PK_H, ConsumePubKey(provider)}};
+        if ("Bz"_mst << typ && ++candidates && !(fragcode--)) return {{Fragment::OLDER, provider.ConsumeIntegralInRange<uint32_t>(1, 0x7FFFFFF)}};
+        if ("Bz"_mst << typ && ++candidates && !(fragcode--)) return {{Fragment::AFTER, provider.ConsumeIntegralInRange<uint32_t>(1, 0x7FFFFFF)}};
+        if ("Bondu"_mst << typ && ++candidates && !(fragcode--)) return {{Fragment::SHA256, PickValue(provider, TEST_DATA.sha256)}};
+        if ("Bondu"_mst << typ && ++candidates && !(fragcode--)) return {{Fragment::RIPEMD160, PickValue(provider, TEST_DATA.ripemd160)}};
+        if ("Bondu"_mst << typ && ++candidates && !(fragcode--)) return {{Fragment::HASH256, PickValue(provider, TEST_DATA.hash256)}};
+        if ("Bondu"_mst << typ && ++candidates && !(fragcode--)) return {{Fragment::HASH160, PickValue(provider, TEST_DATA.hash160)}};
+
+        // Rules for constructing wrappers
+        if ("Wdu"_mst << typ && ++candidates && !(fragcode--)) return {{{"B"_mst | td | tu}, Fragment::WRAP_A}};
+        if ("Wdu"_mst << typ && ++candidates && !(fragcode--)) return {{{"Bo"_mst | td | tu}, Fragment::WRAP_S}};
+        if ("Bondu"_mst << typ && ++candidates && !(fragcode--)) return {{{"K"_mst | to | tn | td}, Fragment::WRAP_C}};
+        if ("Bondu"_mst << typ && ++candidates && !(fragcode--)) return {{{"Vz"_mst}, Fragment::WRAP_D}};
+        if ("Vzon"_mst << typ && ++candidates && !(fragcode--)) return {{{"B"_mst | tz | to | tn}, Fragment::WRAP_V}};
+        if ("Bondu"_mst << typ && ++candidates && !(fragcode--)) return {{{"Bn"_mst | to | tu}, Fragment::WRAP_J}};
+        if ("Bzondu"_mst << typ && ++candidates && !(fragcode--)) return {{{"B"_mst | tz | to | tn | td}, Fragment::WRAP_N}};
+        if ("BVKzonu"_mst << typ && ++candidates && !(fragcode--)) return {{{"V"_mst | tz, base | tz | tu}, Fragment::AND_V}};
+
+        // Rules for constructing connectives
+        if ("Bzondu"_mst << typ && ++candidates && !(fragcode--)) return {{{"B"_mst | tz | td, "W"_mst | tz | td}, Fragment::AND_B}};
+        if ("Bzodu"_mst << typ && ++candidates && !(fragcode--)) return {{{"Bd"_mst | tz, "Wd"_mst | tz}, Fragment::OR_B}};
+        if ("Vzo"_mst << typ && ++candidates && !(fragcode--)) return {{{"Bdu"_mst | tz | to, "V"_mst | tz | toz}, Fragment::OR_C}};
```

Comment:
Why passing `| toz` to the Z parameter of `or_c(X, Z)` here? No rule depend on Z having the `o` property AFAICT.

Reply:
Self answer: it's the other way around. You want Z to have the `z` property only if the parent needs to be either `z` or `o`, therefore the `tz | toz`.